// app/app/dashboard/suppliers/page.tsx
"use client";

import { useState } from "react";
import { Card } from "@/components/ui/card";
import { Button } from "@/components/ui/button";
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from "@/components/ui/table";
import { Badge } from "@/components/ui/badge";
import Link from "next/link";

interface Supplier {
  id: string;
  name: string;
  cnpj: string;
  email: string;
  phone: string;
  type: "ME" | "EPP" | "OUTROS";
  status: "active" | "inactive" | "suspended";
  registrationDate: string;
}

export default function SuppliersPage() {
  const [suppliers] = useState<Supplier[]>([
    {
      id: "1",
      name: "Empresa A Ltda",
      cnpj: "12.345.678/0001-90",
      email: "contato@empresaa.com",
      phone: "(77) 3333-4444",
      type: "ME",
      status: "active",
      registrationDate: "2024-01-15"
    },
    {
      id: "2",
      name: "Empresa B Ltda",
      cnpj: "98.765.432/0001-10",
      email: "contato@empresab.com",
      phone: "(77) 3333-5555",
      type: "EPP",
      status: "active",
      registrationDate: "2024-02-01"
    }
  ]);

  const getStatusBadge = (status: Supplier['status']) => {
    const statusMap = {
      active: { label: "Ativo", variant: "success" as const },
      inactive: { label: "Inativo", variant: "secondary" as const },
      suspended: { label: "Suspenso", variant: "destructive" as const }
    };

    return statusMap[status];
  };

  const getTypeBadge = (type: Supplier['type']) => {
    const typeMap = {
      ME: { label: "Microempresa", variant: "default" as const },
      EPP: { label: "Empresa de Pequeno Porte", variant: "default" as const },
      OUTROS: { label: "Outros", variant: "secondary" as const }
    };

    return typeMap[type];
  };

  return (
    <div className="space-y-6">
      <div className="flex justify-between items-center">
        <h1 className="text-3xl font-bold">Fornecedores</h1>
        <Link href="/dashboard/suppliers/new">
          <Button>Novo Fornecedor</Button>
        </Link>
      </div>

      <Card>
        <div className="p-6">
          <Table>
            <TableHeader>
              <TableRow>
                <TableHead>Razão Social</TableHead>
                <TableHead>CNPJ</TableHead>
                <TableHead>Contato</TableHead>
                <TableHead>Tipo</TableHead>
                <TableHead>Data de Cadastro</TableHead>
                <TableHead>Status</TableHead>
                <TableHead>Ações</TableHead>
              </TableRow>
            </TableHeader>
            <TableBody>
              {suppliers.map((supplier) => (
                <TableRow key={supplier.id}>
                  <TableCell>{supplier.name}</TableCell>
                  <TableCell>{supplier.cnpj}</TableCell>
                  <TableCell>
                    <div>
                      <div>{supplier.email}</div>
                      <div className="text-sm text-gray-500">{supplier.phone}</div>
                    </div>
                  </TableCell>
                  <TableCell>
                    <Badge variant={getTypeBadge(supplier.type).variant}>
                      {getTypeBadge(supplier.type).label}
                    </Badge>
                  </TableCell>
                  <TableCell>
                    {new Date(supplier.registrationDate).toLocaleDateString('pt-BR')}
                  </TableCell>
                  <TableCell>
                    <Badge variant={getStatusBadge(supplier.status).variant}>
                      {getStatusBadge(supplier.status).label}
                    </Badge>
                  </TableCell>
                  <TableCell>
                    <div className="flex gap-2">
                      <Link href={`/dashboard/suppliers/${supplier.id}`}>
                        <Button variant="outline" size="sm">
                          Visualizar
                        </Button>
                      </Link>
                      <Link href={`/dashboard/suppliers/${supplier.id}/documents`}>
                        <Button variant="secondary" size="sm">
                          Documentos
                        </Button>
                      </Link>
                    </div>
                  </TableCell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </div>
      </Card>

      <Card>
        <div className="p-6">
          <h2 className="text-xl font-semibold mb-4">Documentos Pendentes</h2>
          <Table>
            <TableHeader>
              <TableRow>
                <TableHead>Fornecedor</TableHead>
                <TableHead>Documento</TableHead>
                <TableHead>Vencimento</TableHead>
                <TableHead>Status</TableHead>
                <TableHead>Ações</TableHead>
              </TableRow>
            </TableHeader>
            <TableBody>
              <TableRow>
                <TableCell>Empresa A Ltda</TableCell>
                <TableCell>Certidão Negativa Federal</TableCell>
                <TableCell>15/04/2024</TableCell>
                <TableCell>
                  <Badge variant="warning">Próximo ao Vencimento</Badge>
                </TableCell>
                <TableCell>
                  <Button variant="outline" size="sm">
                    Notificar
                  </Button>
                </TableCell>
              </TableRow>
              <TableRow>
                <TableCell>Empresa B Ltda</TableCell>
                <TableCell>Certidão Negativa Municipal</TableCell>
                <TableCell>10/04/2024</TableCell>
                <TableCell>
                  <Badge variant="destructive">Vencido</Badge>
                </TableCell>
                <TableCell>
                  <Button variant="outline" size="sm">
                    Notificar
                  </Button>
                </TableCell>
              </TableRow>
            </TableBody>
          </Table>
        </div>
      </Card>
    </div>
  );
}
