// app/app/api/processes/[id]/bids/route.ts
import { prisma } from "@/lib/prisma"

export const dynamic = "force-dynamic"

export async function GET(
  request: Request,
  { params }: { params: { id: string } }
) {
  try {
    const bids = await prisma.bid.findMany({
      where: {
        processId: params.id
      },
      include: {
        user: true
      },
      orderBy: {
        value: 'asc'
      }
    })
    
    return Response.json(bids)
  } catch (error) {
    console.error('Error fetching bids:', error)
    return Response.json({ error: 'Failed to fetch bids' }, { status: 500 })
  }
}

export async function POST(
  request: Request,
  { params }: { params: { id: string } }
) {
  try {
    const data = await request.json()
    const bid = await prisma.bid.create({
      data: {
        ...data,
        processId: params.id,
        status: "PENDING"
      }
    })
    return Response.json(bid)
  } catch (error) {
    console.error('Error creating bid:', error)
    return Response.json({ error: 'Failed to create bid' }, { status: 500 })
  }
}
