// app/app/dashboard/reports/page.tsx
"use client";

import { Card } from "@/components/ui/card";
import { Button } from "@/components/ui/button";
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from "@/components/ui/select";
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow,
} from "@/components/ui/table";
import { useState } from "react";
import dynamic from 'next/dynamic';
import { Line } from 'react-chartjs-2';
import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend
} from 'chart.js';

ChartJS.register(
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend
);

const Chart = dynamic(() => import('react-chartjs-2').then(mod => mod.Line), {
  ssr: false,
  loading: () => <div>Carregando gráfico...</div>
});

export default function ReportsPage() {
  const [reportType, setReportType] = useState("bids");
  const [period, setPeriod] = useState("month");

  const chartData = {
    labels: ['Jan', 'Fev', 'Mar', 'Abr', 'Mai', 'Jun'],
    datasets: [
      {
        label: 'Licitações Realizadas',
        data: [12, 19, 3, 5, 2, 3],
        fill: false,
        borderColor: 'rgb(75, 192, 192)',
        tension: 0.1,
      },
    ],
  };

  const chartOptions = {
    responsive: true,
    plugins: {
      legend: {
        position: 'top' as const,
      },
      title: {
        display: true,
        text: 'Evolução de Licitações',
      },
    },
  };

  return (
    <div className="space-y-6">
      <div className="flex justify-between items-center">
        <h1 className="text-3xl font-bold">Relatórios</h1>
        <div className="flex gap-4">
          <Select value={reportType} onValueChange={setReportType}>
            <SelectTrigger className="w-[180px]">
              <SelectValue placeholder="Tipo de Relatório" />
            </SelectTrigger>
            <SelectContent>
              <SelectItem value="bids">Licitações</SelectItem>
              <SelectItem value="contracts">Contratos</SelectItem>
              <SelectItem value="suppliers">Fornecedores</SelectItem>
            </SelectContent>
          </Select>
          <Select value={period} onValueChange={setPeriod}>
            <SelectTrigger className="w-[180px]">
              <SelectValue placeholder="Período" />
            </SelectTrigger>
            <SelectContent>
              <SelectItem value="month">Mensal</SelectItem>
              <SelectItem value="quarter">Trimestral</SelectItem>
              <SelectItem value="year">Anual</SelectItem>
            </SelectContent>
          </Select>
          <Button>Gerar Relatório</Button>
        </div>
      </div>

      <div className="grid grid-cols-1 md:grid-cols-3 gap-6">
        <Card className="p-6">
          <h3 className="text-lg font-semibold mb-2">Total de Licitações</h3>
          <p className="text-3xl font-bold text-blue-600">45</p>
          <p className="text-sm text-gray-500">Últimos 6 meses</p>
        </Card>

        <Card className="p-6">
          <h3 className="text-lg font-semibold mb-2">Valor Total Contratado</h3>
          <p className="text-3xl font-bold text-green-600">R$ 2.5M</p>
          <p className="text-sm text-gray-500">Últimos 6 meses</p>
        </Card>

        <Card className="p-6">
          <h3 className="text-lg font-semibold mb-2">Economia Gerada</h3>
          <p className="text-3xl font-bold text-purple-600">R$ 500K</p>
          <p className="text-sm text-gray-500">Últimos 6 meses</p>
        </Card>
      </div>

      <Card className="p-6">
        <h2 className="text-xl font-semibold mb-4">Evolução de Licitações</h2>
        <div className="h-[400px]">
          <Chart data={chartData} options={chartOptions} />
        </div>
      </Card>

      <div className="grid grid-cols-1 md:grid-cols-2 gap-6">
        <Card className="p-6">
          <h2 className="text-xl font-semibold mb-4">Licitações por Modalidade</h2>
          <Table>
            <TableHeader>
              <TableRow>
                <TableHead>Modalidade</TableHead>
                <TableHead>Quantidade</TableHead>
                <TableHead>Valor Total</TableHead>
              </TableRow>
            </TableHeader>
            <TableBody>
              <TableRow>
                <TableCell>Pregão Eletrônico</TableCell>
                <TableCell>25</TableCell>
                <TableCell>R$ 1.500.000,00</TableCell>
              </TableRow>
              <TableRow>
                <TableCell>Concorrência</TableCell>
                <TableCell>10</TableCell>
                <TableCell>R$ 800.000,00</TableCell>
              </TableRow>
              <TableRow>
                <TableCell>Tomada de Preços</TableCell>
                <TableCell>10</TableCell>
                <TableCell>R$ 200.000,00</TableCell>
              </TableRow>
            </TableBody>
          </Table>
        </Card>

        <Card className="p-6">
          <h2 className="text-xl font-semibold mb-4">Principais Fornecedores</h2>
          <Table>
            <TableHeader>
              <TableRow>
                <TableHead>Fornecedor</TableHead>
                <TableHead>Contratos</TableHead>
                <TableHead>Valor Total</TableHead>
              </TableRow>
            </TableHeader>
            <TableBody>
              <TableRow>
                <TableCell>Empresa A Ltda</TableCell>
                <TableCell>5</TableCell>
                <TableCell>R$ 500.000,00</TableCell>
              </TableRow>
              <TableRow>
                <TableCell>Empresa B Ltda</TableCell>
                <TableCell>3</TableCell>
                <TableCell>R$ 300.000,00</TableCell>
              </TableRow>
              <TableRow>
                <TableCell>Empresa C Ltda</TableCell>
                <TableCell>2</TableCell>
                <TableCell>R$ 200.000,00</TableCell>
              </TableRow>
            </TableBody>
          </Table>
        </Card>
      </div>
    </div>
  );
}
