// app/app/api/documents/route.ts
import { prisma } from "@/lib/prisma"
import { DocumentType } from "@prisma/client"

export const dynamic = "force-dynamic"

export async function GET() {
  try {
    const documents = await prisma.document.findMany({
      include: {
        process: true,
      },
    })
    return Response.json(documents)
  } catch (error) {
    console.error('Error fetching documents:', error)
    return Response.json({ error: 'Failed to fetch documents' }, { status: 500 })
  }
}

export async function POST(request: Request) {
  try {
    const formData = await request.formData()
    const file = formData.get('file') as File
    const processId = formData.get('processId') as string | null
    
    const documentData = {
      title: file.name,
      type: DocumentType.NOTICE,
      url: 'https://example.com/documents/placeholder',
      ...(processId && { process: { connect: { id: processId } } }),
    }
    
    const document = await prisma.document.create({
      data: documentData,
    })
    
    return Response.json(document)
  } catch (error) {
    console.error('Error creating document:', error)
    return Response.json({ error: 'Failed to create document' }, { status: 500 })
  }
}
